src/entities/communitySbt.ts

```diff
@@ -359,83 +364,332 @@ class SBT {
-            const data = await client.query<{
-                seasonUserPointzs: SubgraphPointzResponse[]
-            }>({
-                query: gql(pointzQuery),
+        const client = new ApolloClient({
+            cache: new InMemoryCache(),
+            link: new HttpLink({ uri: subgraphUrl, fetch })
+        })
+      
+        const scores: Map<string, number> = new Map<string, number>();
+      
+        let skip = 0;
+        while (true) {
+            const data = await client.query({
+                query: gql(activityQuery),
                 variables: {
-                    id: id,
-                    unit: unit
+                    skipCount: skip,
                 },
             });
-    
-            if (!data?.data?.seasonUserPointzs) {
-                return undefined;
+          skip += 1000;
+      
+          for (const wallet of data.data.wallets) {
+            let score = 0;
+      
+            for (const position of wallet.positions) {
+              const token = position.amm.rateOracle.token.name;
+              const decimals: number = position.amm.rateOracle.token.decimals;
+      
+              const termEnd = Number(
+                ethers.utils.formatUnits(position.amm.termEndTimestamp.toString(), 18),
+              );
+      
+              for (const swap of position.swaps) {
+                const swapTime: number = swap.transaction.createdTimestamp;
+                const swapNotional = Number(
+                  ethers.utils.formatUnits(swap.variableTokenDelta.toString(), decimals),
+                );
+      
+                if (seasonStart < swapTime && swapTime <= seasonEnd) {
+                  const timeWeightedNotional =
+                    (Math.abs(swapNotional) * (termEnd - swapTime)) / ONE_YEAR_IN_SECONDS;
+                  switch (token) {
+                    case 'ETH': {
+                      score += timeWeightedNotional * ethPrice;
+                      break;
+                    }
+                    default: {
+                      score += timeWeightedNotional;
+                    }
+                  }
+                }
+              }
+            }
+      
+            if (score > 0) {
+              scores.set(wallet.id as string, score);
             }
+          }
+      
+          if (data.data.wallets.length < 1000) {
+            break;
+          }
+        }
+        return scores;
+    }
 
-            for (const pointz of data.data.seasonUserPointzs) {
-                if (pointz.seasonUser.id === userId.toLowerCase()) {
-                    const badgeQuery = `
-                        query( $id: String) {
-                            badge(id: $id){
-                                id
-                                mintedTimestamp
+    /**
+   * @dev Query the Main subgraph and retrieve season's liquidity
+   * score of all users based on time weighted liquidity.
+   * Score is based on both mints and swaps.
+   */
+    async getLPScores(
```

Comment:
this seems really similar to: getTraderScores, why are we not reusing code, instead of copy pasting?

Reply:
reused in ffa879f6c11d984ee3999eb2d9ecc6b7195bb293